wcfsetup/install/files/lib/system/language/LanguageFactory.class.php

```diff
@@ -106,8 +88,8 @@ public function getLanguageByCode($languageCode)
             if (isset($row['languageID'])) {
                 return new Language($row['languageID']);
             }
-        } elseif (isset($this->cache['codes'][$languageCode])) {
-            return $this->getLanguage($this->cache['codes'][$languageCode]);
+        } elseif (isset($this->cache->getCache()->codes[$languageCode])) {
```

Comment:
You already added helper methods to the cache, why not use them to handle this instead of manually poking in arrays structures?